src/planning/planner.ts

```diff
@@ -84,6 +86,7 @@ export class Planner implements InspectablePlanner {
     if (inspectorFactory) {
       this.inspector = inspectorFactory.create(this);
     }
+    this.noSpecEx = noSpecEx;
```

Comment:
passing in a null speculator in the planner constructor is enough to disable speculative execution in the planner.ts, there is no need to keep the flag here too.